finality-provider/service/fastsync.go

```diff
@@ -53,6 +58,10 @@ func (fp *FinalityProviderInstance) FastSync(startHeight, endHeight uint64) (*Fa
 			if fp.hasProcessed(b) {
 				continue
 			}
+			// check if it is allowed to send finality
+			if b.Height < activationBlkHeight {
+				continue
+			}
```

Comment:
maybe this check can be moved out of the for loop by comparing the activation height against start height and end height?

Reply:
done at https://github.com/babylonlabs-io/finality-provider/pull/101/commits/e03994d40a2c50373df6793f63c1be78e4e0ec3f